### PR TITLE
CNV-24656: delete policy with confirmation step

### PR DESCRIPTION
**Pull request by upalatucci**
Before there was no delete modal for this policy.
Deleting the resource was not allowed in the UI.

Because deleting the policy can cause issues to nodes, the user can delete the policy only if its archived.

Screenshot from 2023-01-24 11-42-40
Screenshot from 2023-01-24 11-41-13
Screenshot from 2023-01-24 11-40-03

### CODE REVIEW

**Comment by yaacov** [comment]
hi, I'm missing something, I thought the desision was not to delete, and instead add a state in the edit modal that will start a more safe deletion prosses, what am I missing ?

**Comment by upalatucci** [comment]
@yaacov There were some feedback from Dan about that during the demo and we discussed about it a little bit.
If the user know what he is doing, this can be a valuable thing

**Comment by yaacov** [comment]
@dankenigsberg @phoracek hi, just checking that we want to delete it, I thought the desision was to set the spec to `state: absent`

**Comment by dankenigsberg** [comment]
@yaacov these are two very different things. This PR is about deleting an nncp. As the warning text says this leaves existing node config behind. This is what `oc delete nncp` does, and we should educate our users about it 

Actually removing a network device (bridge, vlan, bond) requires setting `absent`. This should be exposed in the GUI too, but on another PR

**Comment by upalatucci** [comment]
[this is the pr](https://github.com/nmstate/nmstate-console-plugin/pull/6) that add the `absent` state btw

**Comment by yaacov** [comment]
@dankenigsberg yes, these are two different actions, my concern is that because we give an easy delete we allow users to ignore the "good path" of setting the state first and just deleting the object (a "avarge customer" just check the checkbox without reading the text, see kubvirt ui customer feedback) , a better aprouch may be to make the `state` setting to `absent` more easy to use by making it a kebab action or disabling delete for components that are not set to `absent` and finish the process, did you discuss other solutions ?

**Comment by dankenigsberg** [comment]
We considered doing something like this in the backend, but rejected it due to its complexity. I don't think that the gui can evaluate whether a network entity has to be absent on all nodes before a specific nncp is deleted.

**Comment by yaacov** [comment]
I'm ok if the desision if it was discussed with the pros and cons of it, my consern is that making it easier (one click + checkbox) to delete, e.g the "bad path" then to set the "state" to "absent" ( go to edit, find the state, set it currently ) will educate people to use the "bad path"
IMHO we can look into this issue and maybe find nicer ways to solve the problem of allowing delete from the ui

**Comment by dankenigsberg** [comment]
imho the frustration about inability to delete an object you have just created is worse.

GUI should make it easier to explore which network devices are available on each worker, which would make it more apparent that a policy is needed to make something absent. But that's a future epic.

**Comment by yaacov** [comment]
I agree we can allow users to delete an object , my concern is that by making it easier to delete the object without removing it first we guide users to go the "bad path" instead of educating Them about how to remove the settings first
IMHO a help text with checkbox has drawbacks that where highlited by the usabilty research of last version, and while we should allow for deleting the object, we should do it in an opinionated way that will adicate users about the current way of first removing the settings

**Comment by upalatucci** [comment]
@yaacov @dankenigsberg updated according to the meeting

File: .github/workflows/on_pull_request.yaml [hunk 1]
```diff
@@ -77,6 +77,7 @@ jobs:
           headed: false
 
       - uses: actions/upload-artifact@v3
+        if: always()
```
Comment (yaacov) [inline .github/workflows/on_pull_request.yaml]: maybe in a different PR ?

Comment (upalatucci) [reply .github/workflows/on_pull_request.yaml]: yes sure

File: src/policies/components/EditModal.tsx [hunk 2]
```diff
@@ -77,14 +79,14 @@ const EditModal: React.FC<EditModalProps> = ({ closeModal, isOpen, policy }) =>
...
-                  variant={'primary'}
+                  variant={ButtonVariant.primary}
                   form="edit-policy-form"
                 >
                   {t('Save')}
                 </Button>
               </ActionListItem>
               <ActionListItem>
-                <Button onClick={closeModal} variant="link">
+                <Button onClick={closeModal} variant={ButtonVariant.secondary}>
```
Comment (yaacov) [inline src/policies/components/EditModal.tsx:89]: a different PR ?

**Review by yaacov — APPROVED** [review]

File: src/policies/components/EditModal.tsx (hunk 2 above)
Comment (upalatucci) [reply src/policies/components/EditModal.tsx:89]: mmmmh... i think its good

**Pull request merged**